File: 01_Array_Hashings/06_Encode_And_Decode_Strings/271_Encode_And_Decode_Strings.py
from typing import List

class Solution:

    def encode(self, strs: List[str]) -> str:
        res = ''
        for s in strs:
            res += str(len(s)) + '#' + s
        return res

    def decode(self, s: str) -> List[str]:
        res = []
        i = 0
        
        while i < len(s):
            s_len = ''
            while s[i] != '#':
                s_len += s[i]
                i += 1
            
            i += 1
            
            length = int(s_len)
            res.append(s[i:i+length])
            
            i += length
            
        return res

File: 01_Array_Hashings/06_Encode_And_Decode_Strings/271_Encode_And_Decode_Strings.cpp
class Solution {
public:

    string encode(vector<string>& strs) {

        string res = "";

        for (string s : strs){
            res.append(to_string(s.length()) +"#"+s);
        }
        cout << res;

        return res;

    }

    vector<string> decode(string s) {
        vector<string> res = {};
        int i = 0;
        while (i < s.length()){
            string s_len = "";

            while(s[i] != '#'){
                s_len += s[i];
                i++;
            }
            i++;
            int len = stoi(s_len);
            res.push_back(s.substr(i, len));
            i += len;
        }

        return res;
    }
};